StayFit/CommunityViewController.m

```diff
@@ -0,0 +1,221 @@
+    [self.fitnessTableView insertSubview:self.refreshControl atIndex:0];
+    self.fitnessTableView.dataSource = self;
+    self.fitnessTableView.delegate = self;
+    [self fitnessCase];
+    
+}
+
+- (IBAction)segact:(id)sender {
+    //checks which segment is active
+    switch (self.segout.selectedSegmentIndex){
+        case 0:
+            [self fitnessCase];
+            break;
+        case 1:
+            [self recipesCase];
+            break;
+        case 2:
+            [self gymBuddy];
+            break;
+    }
+}
+
+-(void) fitnessCase{
+    self.addButton.hidden = false;
+    self.arrayOfFitnessPosts = [[NSMutableArray alloc] init];
+    [self fetchFitnessPosts];
+}
+
+-(void) recipesCase{
+    self.addButton.hidden = false;
+    self.arrayofRecipesPosts = [[NSMutableArray alloc] init];
+    [self fetchRecipesPosts];
+}
+
+-(void) gymBuddy{
+    self.addButton.hidden = true;
+    self.arrayOfGymBuddies = [[NSMutableArray alloc] init];
+    [self fetchGymBuddies];
+}
+
+-(void) fetchPosts {
+    if (self.segout.selectedSegmentIndex == 0) {
+        [self fetchFitnessPosts];
+    } else if (self.segout.selectedSegmentIndex == 1) {
+        [self fetchRecipesPosts];
+    } else {
+        [self fetchGymBuddies];
+    }
+}
+
+-(void) fetchFitnessPosts{
+    //creates a new query for fitness posts
+        PFQuery *postQuery = [PFQuery queryWithClassName:@"Post"];
+        [postQuery orderByDescending:@"createdAt"];
+        [postQuery includeKey:@"author"];
+        postQuery.limit = 20;
+    
+        [postQuery findObjectsInBackgroundWithBlock:^(NSArray *posts, NSError *error) {
+            if (posts) {
+                self.arrayOfFitnessPosts = posts;
+                [self.fitnessTableView reloadData];
+            }
+            else {
+                NSLog(@"%@", error);
+            }
+            [self.refreshControl endRefreshing];
+        }];
+}
+-(void) fetchRecipesPosts{
+    //creates a new query for recipes posts
+    PFQuery *postQuery = [PFQuery queryWithClassName:@"recipesPost"];
+    [postQuery orderByDescending:@"createdAt"];
+    [postQuery includeKey:@"author"];
+    postQuery.limit = 20;
+
+    [postQuery findObjectsInBackgroundWithBlock:^(NSArray *posts, NSError *error) {
+        if (posts.count) {
+            self.arrayofRecipesPosts = posts;
+            [self.fitnessTableView reloadData];
+        }
+        else {
+            NSLog(@"%@", @"no posts");
+        }
+        [self.refreshControl endRefreshing];
+    }];
+}
+
+-(void) fetchGymBuddies{
+    //creates a new query for matching gym buddies
+    PFQuery *postQuery = [PFQuery queryWithClassName:@"_User"];
+    [postQuery orderByDescending:@"createdAt"];
+    [postQuery includeKey:@"author"];
+    [postQuery whereKey:@"fitnessLevel" equalTo:[PFUser currentUser][@"fitnessLevel"]];
+    [postQuery whereKey:@"city" equalTo:[PFUser currentUser][@"city"]];
```

Comment:
good filters, if you choose to allow users to search for other gym buddies at a specific location or level it would be good to pull these parameters out to be set for each use case instead of hard coded to be the same as the user